src/test/ui/macros/macro-multiple-matcher-bindings.rs

```diff
@@ -0,0 +1,25 @@
+// Test that duplicate matcher binding names are caught at declaration time, rather than at macro
+// invocation time.
+//
+// FIXME(mark-i-m): Update this when it becomes a hard error.
+
+// compile-pass
```

Comment:
If you add `// compile-pass` then all the `//~ WARNING` annotations become ignored unfortunately (https://github.com/rust-lang/rust/issues/56277).
Could you use `deny(duplicate_matcher_binding_name)` to get errors instead?